compiler/jsgen.nim

```diff
@@ -2136,6 +2136,11 @@ proc optionaLine(p: Rope): Rope =
   else:
     return p & tnl
 
+proc genAsync(prc: PSym): Rope =
+  var returnType = prc.ast.sons[3][0]
+  if returnType.kind == nkBracketExpr and returnType.sons[0].kind == nkSym and ($returnType.sons[0]) == "Future":
```

Comment:
You need to check against some magic type here. I could have ``otherModule.Future`` and you would transform it to ``async``...

Reply:
Yeah, I wanted to somehow annotate the function in another way, but I wasn't sure how to propagadate it to jsgen.  
Another solution is to generate an impossible name for the function "normal_name@async" and then split it in jsgen, but is it too hacky?

Reply:
Actually one can generate all kinds of names with ``, so that wouldn't work too